recipes/funchook/all/conanfile.py

```diff
@@ -0,0 +1,114 @@
+        "fPIC": True,
+        "disassembler": "diStorm3",
+    }
+
+    _cmake = None
+
+    def export_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            self.copy(patch["patch_file"])
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def package_id(self):
+        self.info.shared_library_package_id()
+
+    def requirements(self):
+        if self.options.disassembler == "capstone":
+            self.requires("capstone/4.0.2")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def validate(self):
+        if self.options.disassembler == "zydis":
+            raise ConanInvalidConfiguration("disassembler 'zydis' currently not supported by this Conan recipe")
+        if 'arm' in str(self.settings.arch):
+            if self.settings.os == "Macos":
+                raise ConanInvalidConfiguration("Funchook does not support ARM on MacOS")
+            if self.options.disassembler != "capstone":
+                raise ConanInvalidConfiguration("disassembler must be 'capstone' for arm arch")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["FUNCHOOK_BUILD_TESTS"] = False
+        self._cmake.definitions["FUNCHOOK_BUILD_SHARED"] = self.options.shared
+        self._cmake.definitions["FUNCHOOK_BUILD_STATIC"] = not self.options.shared
+        if self.options.disassembler == "diStorm3":
+            self._cmake.definitions["FUNCHOOK_DISASM"] = "distorm"
+        else:
+            self._cmake.definitions["FUNCHOOK_DISASM"] = self.options.disassembler
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        self.copy("*.h", dst="include", src=(self._source_subfolder + "/include"))
+        self.copy("*.lib", dst="lib", keep_path=False)
+        self.copy("*.dll", dst="bin", keep_path=False)
+        if self.options.shared:
+            self.copy("*.so*", dst="lib", keep_path=False)
+            self.copy("*.dylib*", dst="lib", keep_path=False)
+        else:
+            self.copy("*.a", dst="lib", keep_path=False)
```

Comment:
```suggestion
        cmake = self._configure_cmake()
        cmake.install()
```

Use cmake install instead https://github.com/kubo/funchook/blob/v1.1.0/CMakeLists.txt#L230

Reply:
When I tried that, I got the .pdb files included in the output package which (I guess) goes against best practices.  I'm hesitant to keep layering on patches on the CMakeLists.txt file unless I have to.  Know of any way to exclude the .pdb files programmatically from here?  

If not, should I keep my original method or should I keep hacking away at patches for CMakeLists.txt?

Reply:
Use cmake.install(), then remove *.pdb and *.cmake